.github/workflows/gotest.yml

```diff
@@ -0,0 +1,18 @@
+name: Run `go test`
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  run-go-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Get version from go.mod
```

Comment:
Can we get the go version the same way that Juju does just to keep it consistent? 

https://github.com/juju/juju/blob/0881dffdd57e535c5cf56e59c9fc897b74411438/.github/workflows/client-tests.yml#L40

Save us from having to trust a small action repo that we don't have much control over.

Reply:
Done